agent/app/agent.go

```diff
@@ -108,6 +110,26 @@ func newAgent(
 		return nil, err
 	}
 
+	var metadataManager containermetadata.MetadataManager
+	if cfg.ContainerMetadataEnabled {
+		// Get the oldest Docker Client version with up to date inspect API
```

Comment:
Do you mean moving it into a method but still within the app package? Maybe I misconstrued your request.